File: sort.rb
a = [*1..1000].sample(200)
b = [*1..1000].sample(150)

a_sort2 = a
b_sort2 = b

puts"a\n"
puts a

puts"\nb\n"
puts b

start = Time.now

c= a.concat(b)
for k in 0..c.size-1 
   for l in 0..c.size-2-k
        if (c[l]>c[l+1]) then
	c[l],c[l+1] = c[l+1],c[l] 
	           
      end   
   end
end
finish = Time.now

puts"\nc\n"
puts c
puts("This is the time it takes to rearrange the numbers in the list combined by bubble sort. ")
puts(finish-start)

puts"\nc_sort2\n"
start = Time.now

c_sort2 = a_sort2.concat(b_sort2)
puts c_sort2.sort

finish = Time.now

puts("measured time when using sort function.")
puts(finish-start)

puts "\noverlap\n"
n = 1
for j in 0..c.size-1 do
    if c[j] == c[j+1] then 
	n=n+1 
    else 
	if n>=2 then 
            puts c[j]
	    puts" "
	    puts n
	    puts" " 
            n=1
         end
    end
end                                     


 

File: sort.go
package main
import "fmt"
import "math/rand"
import "time"
import "sort"

func main(){

var a [200]int
var b [200]int
var a_sort2 [200]int
var b_sort2 [200]int

println("a value")

for  i:= 0;i<200;i++ { 
  a[i] = rand.Intn(2000)
        fmt.Print(a[i]," ")  
        }

println("\nb value")
for j:= 0;j<200;j++{
  b[j] = rand.Intn(2000)
        fmt.Print(b[j]," ")
	}

a_sort2=a
b_sort2=b

fmt.Println()
startTime:=time.Now()

var c []int
var c_sort2 []int
c_sort2=c

for index, _ := range a{
                 c= append(c,a[index])
         }

for index, _ := range b {
                 c = append(c, b[index])
         }

for o := 0;o<len(c)-1;o++{
   for s := 0;s<len(c)-1-o;s++{
      if c[s] > c[s+1] { 
          c[s],c[s+1] = c[s+1],c[s] }
   }
}

elapsedTime := time.Since(startTime)
fmt.Printf("This is the time it takes to rearrange the numbers in the list combined by bubble sort: %s\n", elapsedTime)

startTime2 := time.Now()
for index, _ := range a_sort2{
                 c_sort2= append(c_sort2,a_sort2[index])
         }

for index, _ := range b_sort2 {
                 c_sort2 = append(c_sort2, b_sort2[index])
         }
sort.Ints(c_sort2)
elapsedTime2 := time.Since(startTime2)
fmt.Printf("measured time when using sort function: %s\n", elapsedTime2)

print("\noverlap\n")
n := 1
for j := 0;j < len(c)-1; j++{
    if c[j] == c[j+1] 
	{n=n+1}
    else {
	if n>=2 { 
            fmt.Print(c[j]," ",n," ") 
            n=1}}      
    }
         






